### PR TITLE
SWC-7062

### PR DESCRIPTION
**Pull request by jay-hodgson**
Refactor, and add new UI that uses the same Source Configs hook

### CODE REVIEW

**Comment by jay-hodgson** [comment]
In the new Synapse Homepage:
chrome_Qu9Sih8jCM

Behind a feature flag in all portals!
chrome_9639RGgd84

File: packages/synapse-react-client/src/components/error/ErrorPage.tsx [hunk 1]
```diff
@@ -148,7 +148,7 @@ const ErrorPage: React.FunctionComponent<ErrorPageProps> = props => {
     () => setSendMessageToAdminDialogOpen(false),
     [],
   )
-
+  console.log('Entity ID', entityId)
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/error/ErrorPage.tsx:151]: remove

Comment (nickgros) [inline packages/synapse-react-client/src/components/error/ErrorPage.tsx:151]:
```suggestion
```

File: packages/synapse-types/src/FeatureFlags.ts [hunk 2]
```diff
@@ -15,4 +15,7 @@ export enum FeatureFlagEnum {
 
   // If enabled, show the chatbot entrypoint from the new Synapse Homepage
   HOMEPAGE_CHATBOT = 'HOMEPAGE_CHATBOT',
+
+  // If enabled, show the Portals dropdown on the portal site (to discover other portals)
+  PORTALS_DROPDOWN = 'PORTALS_DROPDOWN',
```
Comment (nickgros) [inline packages/synapse-types/src/FeatureFlags.ts:20]: Please also make sure this is added to the stack builder in https://github.com/Sage-Bionetworks/Synapse-Stack-Builder/blob/develop/src/main/resources/templates/repo/appconfig-config.json

**Review by nickgros — APPROVED** [review]

File: packages/synapse-types/src/FeatureFlags.ts (hunk 2 above)
Comment (jay-hodgson) [reply packages/synapse-types/src/FeatureFlags.ts:20]: https://github.com/Sage-Bionetworks/Synapse-Stack-Builder/pull/662

**Pull request merged**